src/platform/windows/input/key.h

```diff
@@ -6,6 +6,9 @@ struct Key {
     enum class Type { None, Keyboard, MouseMove, MouseButton, ControllerMove, ControllerButton };
     enum class Axis { Invalid = 0, Up = 1, Right, Left, Down };
 
+    std::string button;
```

Comment:
Why are those stored as fields in the struct? Move them to `getButton` / `getDevice`
Just return `std::string` and it'll be fine.

```c++
std::string Key::getDevice() {
    auto device = to_string();
    size_t dpos = device.find('|');
    device = device.substr(0, dpos);
    return device;
}
```